src/components/TodoList.tsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import classNames from 'classnames';
+import { useTodo } from '../providers/TodoProvider';
+import { TodoInfo } from './TodoInfo';
+import { TodoEdit } from './TodoEdit';
+
+export const TodoList = () => {
+  const {
+    visibleTodos,
+    setTodos,
+    modifiedTodo,
+    setModifiedTodo,
+  } = useTodo();
+
+  const handleClick = (todoId: number | null = null) => (
+    (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
+      if (event.detail === 2) {
+        setModifiedTodo(todoId);
+      }
+    }
+  );
+
+  const handleCheck = (todoId: number) => () => {
+    setTodos(prev => (
+      prev.map(todo => {
+        if (todo.id === todoId) {
+          return { ...todo, completed: !todo.completed };
+        }
+
+        return todo;
+      })
+    ));
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {visibleTodos.map(todo => {
+        return (
+          <div
+            key={todo.id}
+            data-cy="Todo"
+            className={classNames('todo', {
+              completed: todo.completed,
+            })}
+            onClick={handleClick(todo.id)}
```

Comment:
It's unnecessary to use a click event and check event.detail === 2. You can simply use a doubleClick event

Reply:
true.